#! /bin/sh /usr/share/dpatch/dpatch-run
## 02_quit.blend_in_.blender.dpatch by Florian Ernst <florian@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Make sure quit.blend gets created, saved and read in $HOME/.blender in
## DP: order to avoid a symlink attack, see bug#298167

@DPATCH@
diff -urNad blender-2.40~/source/blender/blenkernel/intern/blender.c blender-2.40/source/blender/blenkernel/intern/blender.c
--- blender-2.40~/source/blender/blenkernel/intern/blender.c	2005-12-05 16:46:25.000000000 +0100
+++ blender-2.40/source/blender/blenkernel/intern/blender.c	2006-01-09 12:26:32.000000000 +0100
@@ -683,7 +683,7 @@
 	/* no undo state to save */
 	if(undobase.first==undobase.last) return;
 		
-	BLI_make_file_string("/", str, U.tempdir, "quit.blend");
+	BLI_make_file_string("/", str, BLI_gethome(), ".blender/quit.blend");
 
 	file = open(str,O_BINARY+O_WRONLY+O_CREAT+O_TRUNC, 0666);
 	if(file == -1) {
diff -urNad blender-2.40~/source/blender/src/header_info.c blender-2.40/source/blender/src/header_info.c
--- blender-2.40~/source/blender/src/header_info.c	2005-12-19 18:54:17.000000000 +0100
+++ blender-2.40/source/blender/src/header_info.c	2006-01-09 12:24:28.000000000 +0100
@@ -801,7 +801,7 @@
 			
 			strcpy(scestr, G.sce);	/* temporal store */
 			save_over = G.save_over;
-			BLI_make_file_string("/", str, U.tempdir, "quit.blend");
+			BLI_make_file_string("/", str, BLI_gethome(), ".blender/quit.blend");
 			BKE_read_file(str, NULL);
 			G.save_over = save_over;
 			strcpy(G.sce, scestr);
diff -urNad blender-2.40~/source/blender/src/usiblender.c blender-2.40/source/blender/src/usiblender.c
--- blender-2.40~/source/blender/src/usiblender.c	2005-11-23 17:33:19.000000000 +0100
+++ blender-2.40/source/blender/src/usiblender.c	2006-01-09 12:24:57.000000000 +0100
@@ -611,7 +611,7 @@
 
 	if (BLI_exists(tstr)) {
 		char str[FILE_MAXDIR+FILE_MAXFILE];
-		BLI_make_file_string("/", str, U.tempdir, "quit.blend");
+		BLI_make_file_string("/", str, BLI_gethome(), ".blender/quit.blend");
 
 		if(U.uiflag & USER_GLOBALUNDO) BLI_delete(tstr, 0, 0);
 		else BLI_rename(tstr, str);
